e2e_tests/src/lib.rs

```diff
@@ -36,7 +35,7 @@ const TEST_TIMEOUT: Duration = Duration::from_secs(1);
 #[derive(Debug)]
 pub struct TestClient {
     basic_client: BasicClient,
-    created_keys: Option<HashSet<(String, String, ProviderID)>>,
+    created_keys: Option<HashSet<(String, Option<String>, ProviderID)>>,
```

Comment:
Wouldn't it be easier to extend for JWT if you just put `Authentication` here, instead of `Option<String>`?

Reply:
I think the extension would not have to touch those parts as for JWT, there will be similarly than `UnixPeerCredentials` a new variant of `Authentication` without containing any information.

I mean the code would not have to change to work for JWT in the future.

Reply:
Yes, my thought was also that this could be used in the future when there is more than one authenticator and you create keys with them, so you can't just rely on the "default". But that's probs premature optimization.